draft-ietf-mls-architecture.md

```diff
@@ -696,6 +696,109 @@ software is updated while the client is a member of a group. Thus, in addition
+
+MLS relies on the following network services. These network services would
+either need to be the same, or compatible at the wire-format level, in order for
+two different deployments based on them to interoperate.
+
+- An **Authentication Service**, described fully in {{authentication-service}},
+  defines the types of credentials which may be used in a deployment and
+  provides methods for:
+  1. Issuing new credentials,
+  2. Validating existing credentials against a reference identifier, and
+  3. Validating whether or not two credentials represent the same user.
+- A **Delivery Service**, described fully in {{delivery-service}}, provides
+  methods for:
+  1. Delivering messages sent to a group to all members in the group.
+  2. Delivering Welcome messages to new members of a group.
+  3. Downloading KeyPackages for specific clients, and uploading new KeyPackages
+     for a user's own clients.
+- Additional services may or may not be required depending on the application
+  design:
+  - If assisted joining is desired (meaning that the ratchet tree is not
+    provided in Welcome messages), there must be a method to download the
+    ratchet tree corresponding to a group.
+  - If assisted joining is desired and the Delivery Service is not able to
+    compute the ratchet tree itself (because some proposals or commits are sent
+    encrypted), there must be a method for group members to publish the updated
+    ratchet tree after each commit.
+  - If external joiners are allowed, there must be a method to publish a
+    serialized `GroupInfo` object (with an `external_pub` extension) that
+    corresponds to a specific group and epoch, and keep that object in
+    sync with the state of the group.
+  - If an application chooses not to allow assisted or external joining, it may
+    instead provide a method for external users to solicit group members (or a
+    designated service) to add them to a group.
+  - If the application uses external PSKs, or uses resumption PSKs that all
+    members of a group may not have access to, there must be a method for
+    distributing these PSKs to group members.
+  - If an application wishes to detect and possibly discipline members that send
+    malformed commits with the intention of corrupting a group's state, there
+    must be a method for reporting and validating malformed commits.
+
+MLS requires the following parameters to be defined, which must be the same for
+two implementations to interoperate:
+
+- The maximum total lifetime that is acceptable for a KeyPackage.
+- How long to store the resumption secret for past epochs of a group.
+- The degree of tolerance that's allowed for out-of-order message delivery:
+  - How long to keep unused nonce and key pairs for a sender
+  - A maximum number of unused key pairs to keep.
+  - A maximum number of steps that clients will move a secret tree ratchet
+    forward in response to a single message before rejecting it.
```

Comment:
These don't have to be the same, do they?  For example, if the DS doesn't allow out-of-order delivery, then you would never detect a difference.  Maybe just soften the "must be the same", e.g., to "must be compatible"